src/mount.ts

```diff
@@ -20,16 +26,80 @@ export type MountOptions = {
   prepareContainer?: (container: HTMLElement) => void;
 };
 
+export type EnzymePropSelector = {
+  [key: string]: any;
+};
+
+export type ComponentType<Props> =
+  | ComponentClass<Props>
+  | FunctionComponent<Props>;
+
+export type EnzymeSelector = string | ComponentType<any> | EnzymePropSelector;
+
+/**
+ * Inspired by DefinitelyTyped's ReactWrapper.
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cebb88fecfa52f854826e216a537867c11b0150e/types/enzyme/index.d.ts
+ */
+export type ComponentWrapper<P = {}, S = {}, C = Component> = {
+  at(index: number): ComponentWrapper;
+
+  childAt(index: number): ComponentWrapper<any, any>;
+  childAt<P2, S2>(index: number): ComponentWrapper<P2, S2>;
```

Comment:
We _could_ omit the overloads which require types to be specified to work, as they won't benefit us in our tests written in JS. They _might_ still be useful in utility functions in this library though.

Reply:
Let's keep them. I have seen a couple cases in which my IDE was smarter about what props were available thanks to these.

Additionally, if we need them at some point it's easier to use them, whereas if we remove them now, in future we might not even remember these definitions existed.

Reply:
Well, the `childAt` specifically would require the generics to be provided, which as you said, it's not that useful from JS files.

My comment was more about the others where the generic can be inferred.

Reply:
> My comment was more about the others where the generic can be inferred, like find or prop.

To clarify, I was only referring to the overloads where the generics cannot be inferred.